### PR TITLE
Add Migration.kt

### LINKED ISSUE #113: Migration from 0.3.1 to 0.6.0
Just created functions with annotation `@Deprecated` and `replaceWith` expressions to make migration easier in simple cases.

```kotlin
@file:Suppress("PackageDirectoryMismatch", "unused")

package dev.chrisbanes.insetter

import android.view.View

@Deprecated(
    level = DeprecationLevel.ERROR,
    message = "Use applyInsetter instead",
    replaceWith = ReplaceWith(
        "this.applyInsetter { type(statusBars = top, navigationBars = bottom) { padding() } }",
        "dev.chrisbanes.insetter.applyInsetter"
    )
)
fun View.applySystemWindowInsetsToPadding(top: Boolean = false, bottom: Boolean = false) {
    error("should not be used")
}

@Deprecated(
    level = DeprecationLevel.ERROR,
    message = "Use applyInsetter instead",
    replaceWith = ReplaceWith(
        "this.applyInsetter { type(statusBars = top, navigationBars = bottom) { margin() } }",
        "dev.chrisbanes.insetter.applyInsetter"
    )
)
fun View.applySystemWindowInsetsToMargin(top: Boolean = false, bottom: Boolean = false) {
    error("should not be used")
}

object Insetter {
    @Deprecated(
        level = DeprecationLevel.ERROR,
        message = "Use WindowCompat.setDecorFitsSystemWindows() instead",
        replaceWith = ReplaceWith(
            "WindowCompat.setDecorFitsSystemWindows(window, !enabled)",
            "androidx.core.view.WindowCompat"
        )
    )
    fun setEdgeToEdgeSystemUiFlags(view: View, enabled: Boolean = true) {
        error("should not be used")
    }
}
```

### PR DESCRIPTION
**Pull request by osipxd**
Closes #113

### CODE REVIEW

File: library/src/main/java/dev/chrisbanes/insetter/Insetter.kt [hunk 1]
```diff
@@ -536,6 +536,18 @@ class Insetter private constructor(
         fun builder(): Builder = Builder()
 
         internal const val TAG = "Insetter"
+
+        @Deprecated(
+            level = DeprecationLevel.ERROR,
+            message = "Use WindowCompat.setDecorFitsSystemWindows() instead",
+            replaceWith = ReplaceWith(
+                "WindowCompat.setDecorFitsSystemWindows(window, !enabled)",
```
Comment (chrisbanes) [inline library/src/main/java/dev/chrisbanes/insetter/Insetter.kt:544]: I avoided this replacement, as there's no guarantee that `window` will be in the scope

File: library/src/main/java/dev/chrisbanes/insetter/Migration.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+@Deprecated(
+    level = DeprecationLevel.ERROR,
+    message = "Use applyInsetter instead",
+    replaceWith = ReplaceWith(
+        "applyInsetter { type(statusBars = top, navigationBars = bottom) { padding() } }",
+        "dev.chrisbanes.insetter.applyInsetter"
+    )
+)
+fun View.applySystemWindowInsetsToPadding(top: Boolean = false, bottom: Boolean = false) {
```
Comment (chrisbanes) [inline library/src/main/java/dev/chrisbanes/insetter/Migration.kt:31]: Nit: inline all of these functions

Comment (osipxd) [reply library/src/main/java/dev/chrisbanes/insetter/Migration.kt]: Done

File: library/src/main/java/dev/chrisbanes/insetter/Insetter.kt (hunk 1 above)
Comment (osipxd) [reply library/src/main/java/dev/chrisbanes/insetter/Insetter.kt]: Good point!I've removed the replacement statement and left only the function with a comment and ERROR deprecation level.

**Comment by chrisbanes** [comment]
Thanks 🙌

**Review by chrisbanes — APPROVED** [review]

**Comment by chrisbanes** [comment]
Looks like there's some code style issues. Please run: `./gradlew spotlessApply apiDump` and commit the changes.

**Comment by osipxd** [comment]
Hmm... I've added `@Supress("unused")` but looks like the compiler with the flag `-Werror` ignores it.

**Comment by chrisbanes** [comment]
> Hmm... I've added `@Supress("unused")` but looks like the compiler with the flag `-Werror` ignores it.
> Do you know any way to ignore warnings in this code?

The problem is that the params aren't used. `@Suppress("unused")` only suppresses the fact that function isn't used.

You can fix it by using `@Suppress("unused", "UNUSED_PARAMETER")` (I think)

**Comment by osipxd** [comment]
Yes, you're right, it works.

**Review by chrisbanes — APPROVED** [review]

**Pull request merged**